### ISSUE TITLE
PatternVariableAssignment fails when variable used inside an assignment target expression

### ISSUE THREAD

**Issue by delanym** [checkstyle/checkstyle#19750]
PatternVariableAssignment fails on this code

```java
    if (object instanceof Integer integer1) {
      sessions[integer1] = null;
    }
```

**Comment by lithops-zty** [comment]
I can confirm that this is a false positive. Here is a bug report in required format:

I have read check documentation: https://checkstyle.org/checks/coding/patternvariableassignment.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java 
/var/tmp $ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="PatternVariableAssignment"/>
    </module>
</module>
/var/tmp $ cat Test.java 
class Test {
    void foo(Object object, String... sessions) {
        if (object instanceof Integer integer1) {
            sessions[integer1] = null;
        }
    }
}
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"                                         
/var/tmp $ java $RUN_LOCALE -jar checkstyle-13.4.0-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /mnt/d/IntellijProjects/checkstyle-playground/inputs/pattern-variable-assignment/Test.java:4:22: Assignment of pattern variable 'integer1' is not allowed. [PatternVariableAssignment]
Audit done.
Checkstyle ends with 1 errors.
```
---

**Describe what you expect in detail.**

There should be no violations since the pattern variable `integer1` is read, not assigned.

**Comment by romani** [comment]
@delanym , next time you report issue please follow this format.

**Comment by romani** [comment]
fix is done by @ayushactiveat . 
Sorry for later merge.
